site/src/pages/ManagementSettingsPage/OrganizationMembersPage.tsx

```diff
@@ -52,45 +54,96 @@ const OrganizationMembersPage: FC = () => {
 	const organization = organizations?.find((o) => o.name === organizationName);
 	const permissionsQuery = useQuery(organizationPermissions(organization?.id));
 
+	const [memberToDelete, setMemberToDelete] =
+		useState<OrganizationMemberWithUserData>();
+
 	const permissions = permissionsQuery.data;
 	if (!permissions) {
 		return <Loader />;
 	}
 
 	return (
-		<OrganizationMembersPageView
-			allAvailableRoles={organizationRolesQuery.data}
-			canEditMembers={permissions.editMembers}
-			error={
-				membersQuery.error ??
-				addMemberMutation.error ??
-				removeMemberMutation.error ??
-				updateMemberRolesMutation.error
-			}
-			isAddingMember={addMemberMutation.isLoading}
-			isUpdatingMemberRoles={updateMemberRolesMutation.isLoading}
-			me={me}
-			members={members}
-			groupsByUserId={groupsByUserIdQuery.data}
-			addMember={async (user: User) => {
-				await addMemberMutation.mutateAsync(user.id);
-				void membersQuery.refetch();
-			}}
-			removeMember={async (member: OrganizationMemberWithUserData) => {
-				await removeMemberMutation.mutateAsync(member.user_id);
-				void membersQuery.refetch();
-			}}
-			updateMemberRoles={async (
-				member: OrganizationMemberWithUserData,
-				newRoles: string[],
-			) => {
-				await updateMemberRolesMutation.mutateAsync({
-					userId: member.user_id,
-					roles: newRoles,
-				});
-			}}
-		/>
+		<>
+			<OrganizationMembersPageView
+				allAvailableRoles={organizationRolesQuery.data}
+				canEditMembers={permissions.editMembers}
+				error={
+					membersQuery.error ??
+					addMemberMutation.error ??
+					removeMemberMutation.error ??
+					updateMemberRolesMutation.error
+				}
+				isAddingMember={addMemberMutation.isLoading}
+				isUpdatingMemberRoles={updateMemberRolesMutation.isLoading}
+				me={me}
+				members={members}
+				groupsByUserId={groupsByUserIdQuery.data}
+				addMember={async (user: User) => {
+					await addMemberMutation.mutateAsync(user.id);
+					void membersQuery.refetch();
+				}}
+				removeMember={setMemberToDelete}
+				updateMemberRoles={async (
+					member: OrganizationMemberWithUserData,
+					newRoles: string[],
+				) => {
+					await updateMemberRolesMutation.mutateAsync({
+						userId: member.user_id,
+						roles: newRoles,
+					});
+				}}
+			/>
+
+			<ConfirmDialog
+				type="delete"
+				open={memberToDelete !== undefined}
+				onClose={() => setMemberToDelete(undefined)}
+				title="Remove member"
+				confirmText="Remove"
+				onConfirm={async () => {
+					try {
+						if (memberToDelete) {
+							await removeMemberMutation.mutateAsync(memberToDelete?.user_id);
+						}
+						setMemberToDelete(undefined);
+						await membersQuery.refetch();
+						displaySuccess("User removed from organization successfully!");
+					} catch (error) {
+						setMemberToDelete(undefined);
+						displayError(
+							getErrorMessage(error, "Failed to remove user from organization"),
+						);
+					}
```

Comment:
```suggestion
						await membersQuery.refetch();
						displaySuccess("User removed from organization successfully!");
					} catch (error) {
						displayError(
							getErrorMessage(error, "Failed to remove user from organization"),
						);
					} finally {
						setMemberToDelete(undefined);
					}
```

minor, but this is the kind of thing that `finally` can be nice for: unconditional cleanup